service/create/service.go

```diff
@@ -198,13 +199,47 @@ func (s *Service) Boot() {
 	})
 }
 
-func (s *Service) runMachines(machines []node.Node, clusterName, prefix string) error {
-	for no, machine := range machines {
+func (s *Service) runMachines(spec awstpr.Spec, clusterName, prefix string) error {
+	var (
+		machines    []node.Node
+		awsMachines []awsinfo.Node
+	)
+
+	switch prefix {
+	case prefixMaster:
+		machines = spec.Cluster.Masters
+		awsMachines = spec.AWS.Masters
+	case prefixWorker:
+		machines = spec.Cluster.Workers
+		awsMachines = spec.AWS.Workers
+	}
+
+	// TODO(nhlfr): Create a spparate module for validating specs and exeture on the earlier stages.
+	if len(machines) != len(awsMachines) {
+		return microerror.MaskAny(fmt.Errorf("mismatched number of %q machines in the 'spec' and 'aws' sections: %d != %d",
+			prefix,
+			len(machines),
+			len(awsMachines)))
+	}
+
+	cloudConfig, err := s.cloudConfig(prefix)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(machines); i++ {
+		name := fmt.Sprintf("%s-%d", prefix, i)
```

Comment:
Which implications does the naming scheme here have in AWS? I can imagine doing it like this we could create the same name multiple times. Is that correct? 

Further, when we have to manage a naming scheme like this it would be good to define the scheme format as a constant and provide some useful godoc for it.